plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -278,6 +278,79 @@ export const createSessionRecordingEvent = (
 
     return data
 }
+
+export interface SummarizedSessionRecordingEvent {
+    uuid: string
+    timestamp: string
+    team_id: number
+    distinct_id: string
+    session_id: string
+    window_id: string | undefined
+    created_at: string
+    url: string | undefined
+    click_count: number
+    keypress_count: number
+    mouse_activity_count: number
+}
+
+interface RRWebEventSummaryData {
+    href?: string
+    source?: number
+}
+
+interface RRWebEventSummary {
+    timestamp: number
+    type: number
+    data: RRWebEventSummaryData
+}
+
+export const createSessionReplayEvent = (
+    uuid: string,
+    team_id: number,
+    distinct_id: string,
+    timestamp: DateTime,
+    ip: string | null,
+    properties: Properties
+) => {
+    const timestampString = castTimestampOrNow(timestamp, TimestampFormat.ClickHouse)
+
+    const eventsSummaries: RRWebEventSummary[] = properties['$snapshot_data']?.['events_summary'] || []
+    let clickCount = 0
+    let keypressCount = 0
+    let mouseActivity = 0
+    let url: string | undefined = undefined
+    eventsSummaries.forEach((eventSummary: RRWebEventSummary) => {
+        if (eventSummary.type === 3) {
+            mouseActivity += 1
+            if (eventSummary.data.source === 2) {
+                clickCount += 1
+            }
+            if (eventSummary.data.source === 5) {
+                keypressCount += 1
+            }
+        }
+        if (!!eventSummary.data.href?.trim().length && url === undefined) {
```

Comment:
I never know if this is the case but isn't the missing ? after trim meaning that it could throw if there is no href?

Also why not use truthiness and keep it simple :D 
```suggestion
        if (!url && eventSummary.data.href?.trim()) {
```

Reply:
The elvis operator says if the thing is there carry on if not return undefined

<img width="514" alt="Screenshot 2023-04-26 at 09 15 57" src="https://user-images.githubusercontent.com/984817/234513528-b6a72e9e-44ed-486a-92f5-0cdd7ed6a576.png">

I prefer the explicit `!!` to remind myself I'm using truthiness but no strong opinion. I'll see what the tests think of the suggestion and go from there :)